provider.go

```diff
@@ -7,6 +7,9 @@ type Provider interface {
 	// Name returns the provider name.
 	Name() string
 
+	// Id returns the project (GCP) or profile (AWS) or subscription (Azure) for this provider.
+	Id() string
```

Comment:
In Go, the [_rule_](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) is that `ID` should be all uppercase.